packages/office-ui-fabric-react/src/components/TextField/examples/TextField.CustomRender.Example.tsx

```diff
@@ -2,77 +2,81 @@ import * as React from 'react';
 
-export class TextFieldCustomRenderExample extends React.Component<
-  {},
-  {
-    isCalloutVisible: boolean;
-  }
-> {
-  private _iconButtonElement: HTMLElement;
   private _descriptionId: string = getId('description');
+  private _iconButtonId: string = getId('iconButton');
 
-  constructor(props: {}) {
-    super(props);
+  public render(): JSX.Element {
+    return (
+      <Stack gap={20} maxWidth={300}>
+        <TextField label="Custom label rendering" onRenderLabel={this._onRenderLabel} description="Click the (i) icon!" />
 
-    this.state = {
-      isCalloutVisible: false
-    };
+        <TextField
+          label="Custom description rendering"
+          description="A colorful description!"
+          onRenderDescription={this._onRenderDescription}
+        />
+      </Stack>
+    );
   }
 
-  public render(): JSX.Element {
+  private _onRenderDescription = (props: ITextFieldProps): JSX.Element => {
+    const theme = getTheme();
     return (
-      <div className="docs-TextFieldExample">
-        <TextField onRenderLabel={this._onRenderLabel} />
-      </div>
+      <Text variant="small" styles={{ root: { color: theme.palette.green, fontWeight: FontWeights.bold } }}>
+        {props.description}
+      </Text>
     );
-  }
+  };
 
   private _onRenderLabel = (props: ITextFieldProps): JSX.Element => {
-    const { isCalloutVisible } = this.state;
     return (
-      <div className="ms-CustomRenderExample" style={{ display: 'flex', alignItems: 'center' }}>
-        <span>TextField with custom label render</span>
-        <span className="ms-CustomRenderExample-labelIconArea" ref={menuButton => (this._iconButtonElement = menuButton!)}>
-          <IconButton iconProps={{ iconName: 'Info' }} title="Info" ariaLabel="Info" onClick={this._onClick} />
-        </span>
-        {isCalloutVisible && (
+      <>
+        <Stack horizontal verticalAlign="center">
+          <span>{props.label}</span>
+          <IconButton
+            id={this._iconButtonId}
+            iconProps={{ iconName: 'Info' }}
+            title="Info"
+            ariaLabel="Info"
+            onClick={this._onIconClick}
+            styles={{ root: { marginBottom: -3 } }}
+          />
+        </Stack>
+        {this.state.isCalloutVisible && (
           <Callout
-            className="ms-CustomRenderExample-callout"
-            target={this._iconButtonElement}
+            target={'#' + this._iconButtonId}
             setInitialFocus={true}
             onDismiss={this._onDismiss}
             ariaDescribedBy={this._descriptionId}
             role="alertdialog"
           >
-            <div className={calloutChildren} id={this._descriptionId}>
-              In addition to the label itself, this label includes an IconButton which pops out more information in a Callout
-            </div>
-            <div className={calloutChildren}>
+            <Stack gap={20} horizontalAlign="start" maxWidth={300} styles={{ root: { padding: 20 } }}>
```

Comment:
Iirc the default for `horizontalAlign` is `start`, so you might be able to take that out from this line and have the same behavior.